charts/kubeclarity/templates/_helpers.tpl

```diff
@@ -77,3 +77,17 @@ Sets extra Kubeclarity server Service annotations
     {{- end }}
   {{- end }}
 {{- end -}}
+
+{{/*
+Sets extra Kubeclarity server Service annotations
+*/}}
+{{- define "kubeclarity.noproxy" -}}
+{{- $noproxy := list (print (include "kubeclarity.name" .) "." .Release.Namespace ":" (index .Values "kubeclarity-runtime-scan" "resultServicePort")) -}}
+{{- if index .Values "kubeclarity-grype-server" "enabled" -}}
+{{- $noproxy = append $noproxy (print (include "kubeclarity.grype-server.name" .) "." .Release.Namespace ":" (index .Values "kubeclarity-grype-server" "servicePort")) -}}
+{{- end -}}
+{{- if index .Values "kubeclarity-trivy-server" "enabled" -}}
+{{- $noproxy = append $noproxy (print ((index .Values "kubeclarity-trivy-server" "service" "name") | default (include "trivy.fullname" (index .Subcharts "kubeclarity-trivy-server"))) "." .Release.Namespace ":" (index .Values "kubeclarity-trivy-server" "service" "port")) -}}
+{{- end}}
```

Comment:
This variable will get recreated every time this sub template is included, if we define it outside of "kubeclarity.noproxy" then it'll only get created once the first time the templates are loaded. As the data is static I don't think we need to regenerate the list each time. I think we could re-write this like:

```
{{- $noproxy := list (print (include "kubeclarity.name" .) "." .Release.Namespace ":" (index .Values "kubeclarity-runtime-scan" "resultServicePort")) -}}
{{- if index .Values "kubeclarity-grype-server" "enabled" -}}
{{- $noproxy = append $noproxy (print (include "kubeclarity.grype-server.name" .) "." .Release.Namespace ":" (index .Values "kubeclarity-grype-server" "servicePort")) -}}
{{- end -}}
{{- if index .Values "kubeclarity-trivy-server" "enabled" -}}
{{- $noproxy = append $noproxy (print ((index .Values "kubeclarity-trivy-server" "service" "name") | default (include "trivy.fullname" (index .Subcharts "kubeclarity-trivy-server"))) "." .Release.Namespace ":" (index .Values "kubeclarity-trivy-server" "service" "port")) -}}
{{- end}}

{{- define "kubeclarity.noproxy" -}}
{{ join "," $noproxy }}
{{- end -}}
```

and then have in the main template:

```
- name: NO_PROXY
   value: {{ include "kubeclarity.noproxy" . }}
```

or even drop the sub-template and just do this:

```
- name: NO_PROXY
   value: {{ join "," $noproxy }}
```

Reply:
@sambetts-cisco i removed the helper function and added the parsing of the $noproxy to the scanner-template-configmap it should now only be evaluated once for all 4 instances it is being used.